src/components/Paginator.jsx

```diff
@@ -33,27 +56,38 @@ const Paginator = ( {pageNumber, setPageNumber, numberOfPages, getNumberOfPages}
     return ( 
       <div> 
         <button className={visibleButton} onClick={prevPage}>prev</button>
-          Page {pageNumber} out of {numberOfPages}
+          Page {pageNumber} out of {numPages}
         <button className={invisibleButton}></button>
       </div> 
     )
   }
 
-  React.useEffect(() => {
-    getNumberOfPages();
-  }, [pageNumber]);
-
-  return (
-    <div className={centerButtons}>
+  const FullPaginator = () => {
+    return (
+      <div className={centerButtons}>
       {pageNumber == 1 ? 
-        <FirstPagePaginator/>
-      : <div className={centerButtons}>
-          {pageNumber == numberOfPages ?
-            <LastPagePaginator/>
+        numPages == 1 ? 
```

Comment:
all these `*PagePaginator` elements are complicating for every possible case.
the logic can be simplified directly in the return statement:

```jsx
return (
  !loading &&
  <div className={centerButtons}>
    {pageNumber == 1 ?
      <button className={invisibleButton}></button>
      : <button className={visibleButton} onClick={prevPage}>prev</button>
    }
        Page {pageNumber} out of {numPages}
    {pageNumber == numPages ?
      <button className={invisibleButton}></button>
      : <button className={visibleButton} onClick={nextPage}>next</button>
    }
  </div>
)
```

Reply:
fixed- thanks!